src/modules/thresholdsig/example.c

```diff
@@ -0,0 +1,193 @@
+            return 0;
+        }
+    }
+    /* Communication round 3: Exchange partial signatures */
+    for (i = 0; i < N_SIGNERS; i++) {
+        for (j = 0; j < N_SIGNERS; j++) {
+            /* To check whether signing was successful, it suffices to either verify
+             * the the combined signature with the combined public key using
+             * secp256k1_schnorrsig_verify, or verify all partial signatures of all
+             * signers individually. Verifying the combined signature is cheaper but
+             * verifying the individual partial signatures has the advantage that it
+             * can be used to determine which of the partial signatures are invalid
+             * (if any), i.e., which of the partial signatures cause the combined
+             * signature to be invalid and thus the protocol run to fail. It's also
+             * fine to first verify the combined sig, and only verify the individual
+             * sigs if it does not work.
+             */
+            if (!secp256k1_thresholdsig_partial_sig_verify(ctx, &musig_session[i], signer_data[i], N_SIGNERS, j, &partial_sig[j], &pubkeys[indices[j]])) {
+                return 0;
+            }
+        }
+    }
+    return secp256k1_musig_partial_sig_combine(ctx, &musig_session[0], sig, partial_sig, N_SIGNERS);
+}
+
+int main(void) {
+    secp256k1_context* ctx;
+    int i;
+    unsigned char seckeys[N_SIGNERS_TOTAL][32];
+    secp256k1_pubkey pubkeys[N_SIGNERS_TOTAL];
+    secp256k1_pubkey signing_pubkeys[N_SIGNERS_TOTAL];
+    secp256k1_thresholdsig_keyshard shards[N_SIGNERS_TOTAL][N_SIGNERS_TOTAL];
+    secp256k1_pubkey pubcoeff[N_SIGNERS_TOTAL][N_SIGNERS];
+    secp256k1_pubkey combined_pk;
+    unsigned char msg[32] = "this_could_be_the_hash_of_a_msg!";
+    unsigned char pk_hash[32];
+    secp256k1_schnorrsig sig;
+
+    /* Create a context for signing and verification */
+    ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
+    printf("Creating key pairs......");
+    for (i = 0; i < N_SIGNERS_TOTAL; i++) {
+        if (!create_key(ctx, seckeys[i], &pubkeys[i])) {
+            printf("FAILED\n");
+            return 1;
+        }
+    }
+    printf("ok\n");
+    printf("Combining public keys...");
+    if (!secp256k1_musig_pubkey_combine(ctx, NULL, &combined_pk, pk_hash, pubkeys, N_SIGNERS_TOTAL)) {
+        printf("FAILED\n");
+        return 1;
+    }
+    printf("ok\n");
+
+    printf("Sharding keys...");
+    for (i = 0; i < N_SIGNERS_TOTAL; i++) {
+        if (!secp256k1_thresholdsig_keysplit(ctx, shards[i], pubcoeff[i], seckeys[i], N_SIGNERS, N_SIGNERS_TOTAL)) {
+            printf("FAILED\n");
+            return 1;
+        }
+    }
+    printf("ok\n");
+    printf("Verifying and combining shards...");
+    for (i = 0; i < N_SIGNERS_TOTAL; i++) {
+        size_t j;
+        /* Note that on every iteration, the inner loop will overwrite the `signing_pubkeys` array */
```

Comment:
I know the `threshold.md` and `secp256k1_thresholdsig.h` state that the `seckeys[i]` is modified here too.
But wouldn't it be nice to mention it here again, since we have a comment anyway?
Maybe something like ``Note that on every iteration, the inner loop will overwrite seckeys[i] and the `signing_pubkeys` array``?